weave/weave_init.py

```diff
@@ -155,3 +157,21 @@ def init_local() -> InitializedClient:
     server.setup_tables()
     client = weave_client.WeaveClient("none", "none", server)
     return InitializedClient(client)
+
+
+def check_wandb_run_matches_weave_project_entity(
+    weave_entity: str, weave_project: str
+) -> None:
+    wandb_run_id = weave_client.safe_current_wb_run_id()
+    if wandb_run_id:
+        # ex: "entity/project/run_id"
+        wandb_entity, wandb_project, _ = wandb_run_id.split("/")
+        if wandb_project != weave_project:
```

Comment:
TIOLI/pedantry warning

When entity name or project name don't match, it indicates the same underlying problem: the target project does not match.  In general we should make an effort to make our user-facing error messages as helpful as possible, so here we should communicate the full failed requirement not just the local problem.  As-is, the user might input `foo/bar` and `baz/qux`, and they'd get the first error and maybe think to change it to `foo/bar` and `baz/bar`, then run into the second error.  Instead, let's give them all the relevant info at once and explain the underlying problem, to maximize the chances they can understand the problem and fix it in one go.

Reply:
good catch, happy to do that.

Reply:
@nickpenaranda Want to check the copy before I merge?

Reply:
looks good, thank you